pybedlite/overlap_detector.py

```diff
@@ -267,9 +384,9 @@ def from_bed(cls, path: Path) -> "OverlapDetector":
         Returns:
             An overlap detector for the regions in the BED file.
         """
-        detector = OverlapDetector()
+        detector: OverlapDetector[BedRecord] = OverlapDetector()
 
         for region in BedSource(path):
-            detector.add(Interval.from_bedrecord(region))
+            detector.add(region)
```

Comment:
No more converter! It is worth noting this is technically a backwards incompatible change since the returned OverlapDetector will no longer yield `Interval` objects, but now `BedRecord` objects.